### PR TITLE
Add warning to velero version cmd.

### LINKED ISSUE #3017: Add a warning to the `velero version` cmd when versions don't match
I've been seeing a few issues opened with a mismatch in version between Velero client and server. I think there's sufficient coupling between the two that it (the mismatch) could originate the problems being reported.

The `velero version` command should verify any mismatch and warn. Not sure how strong this warning should be. Should it be: "will not work, please update [whichever has the lower version]"?

### PR DESCRIPTION
**Pull request by cvhariharan**
Fixes #3017 

This PR adds a warning to`velero version` command. The warning is shown if the major and minor versions of the client and server mismatch.

If the client version is less than the server version
```
# WARNING: the client version does not match the server version. Please update the client
```
If the server version is less than the client version
```
# WARNING: the client version does not match the server version. Please update the server
```

### CODE REVIEW

File: pkg/cmd/cli/version/version.go
Comment (nrb) [inline pkg/cmd/cli/version/version.go:2]: This should actually be just 2020, if you wouldn't mind updating it.

I know many of the files are inconsistent with this, we're slowly updating them.

**Review by nrb — CHANGES_REQUESTED** [review]
Thanks for this! I think it looks pretty good, I just have one small thing that should change.

File: pkg/cmd/cli/version/version.go
Comment (zubron) [inline pkg/cmd/cli/version/version.go:85]: I would move this check to the end of the function, after the server version has been printed. If there is a client/server mismatch, we would return from this function without telling the user what  the server version actually was and they wouldn't know what version they need to upgrade to.

**Review by zubron — CHANGES_REQUESTED** [review]

File: pkg/cmd/cli/version/version.go
Comment (cvhariharan) [reply pkg/cmd/cli/version/version.go:85]: Yeah, that makes sense.

**Review by nrb — APPROVED** [review]

**Review by ashish-amarnath — APPROVED** [review]

**Comment by ashish-amarnath** [comment]
Merging as the requested changes have been made.
Thank you @cvhariharan for this PR!

**Comment by ashish-amarnath** [comment]
GH isn't allowing merge w/o dismissing @zubron's review. Going to let @zubron do the merge

**Review by zubron — APPROVED** [review]
Oh, sorry for holding this up! Thanks for making the suggested change and for your contribution @cvhariharan! :smile:

**Pull request merged**
